pkg/openservicebroker/server/apiserver.go

```diff
@@ -87,11 +90,55 @@ func (c completedTemplateServiceBrokerConfig) New(delegationTarget genericapiser
 		GenericAPIServer: genericServer,
 	}
 
-	broker := templateservicebroker.DeprecatedNewBrokerInsideAPIServer(
-		c.PrivilegedKubeClientConfig,
-		c.TemplateInformers.Template().InternalVersion().Templates(),
-		c.TemplateNamespaces,
-	)
+	inCluster := false
+	var broker *templateservicebroker.Broker
+	// TODO, this block drops out after the server is moved.
+	if c.PrivilegedKubeClientConfig != nil {
+		broker = templateservicebroker.DeprecatedNewBrokerInsideAPIServer(
+			*c.PrivilegedKubeClientConfig,
+			c.TemplateInformers.Template().InternalVersion().Templates(),
+			c.TemplateNamespaces,
+		)
+		// make sure no one else uses it
+		c.TemplateInformers = nil
+
+	} else {
+		// we're running in cluster
+		// in this case we actually want to construct our own template informer
+		// eventually the config value drops out as it isn't supported server side
```

Comment:
> the comment says "eventually the config value drops out". What is meant by eventually, are you referring to a future evolution of this code(my original interpretation),

Future evolution removes this code.  See https://github.com/openshift/origin/pull/15684/commits/e27e5d5d252e655485950d13c367e51e147b6be5#diff-e73d057d1e253024c42b0fec37e12420L59